src/webgpu/api/validation/encoding/cmds/buffer_texture_copies.spec.ts

```diff
@@ -5,7 +5,88 @@ copy_between_linear_data_and_texture or destroyed,*.
 TODO: plan
 `;
 
+import { poptions, params } from '../../../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { kDepthStencilFormats } from '../../../../capability_info.js';
 import { ValidationTest } from '../../validation_test.js';
 
 export const g = makeTestGroup(ValidationTest);
+
+g.test('depth_stencil_format,copy_usage_and_aspect')
+  .desc(
+    `
+  Validate the combination of usage and aspect of each depth stencil format in copyBufferToTexture
+  and copyTextureToBuffer. See https://gpuweb.github.io/gpuweb/#depth-formats for more details.
+  TODO(jiawei.shao@intel.com): add tests on depth16norm when it is supported.
+`
+  )
+  .cases(
+    params()
+      .combine(poptions('format', kDepthStencilFormats))
+      .combine(poptions('aspect', ['all', 'depth-only', 'stencil-only'] as const))
+  )
+  .fn(async t => {
+    const { format, aspect } = t.params;
+
+    const textureSize = { width: 1, height: 1, depth: 1 };
+    const texture = t.device.createTexture({
+      size: textureSize,
+      format,
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.COPY_DST,
+    });
+
+    const buffer = t.device.createBuffer({
+      size: 32,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+    });
+
+    const emptyStringArray: string[] = [];
+
+    const kDepthStencilFormatCapabilityInBufferTextureCopy = {
+      // kUnsizedDepthStencilFormats
+      depth24plus: {
+        AspectsSupportedInB2T: emptyStringArray,
+        AspectsSupportedInT2B: emptyStringArray,
+      },
+      'depth24plus-stencil8': {
+        AspectsSupportedInB2T: ['stencil-only'],
+        AspectsSupportedInT2B: ['stencil-only'],
+      },
+
+      // kSizedDepthStencilFormats
+      depth32float: {
+        AspectsSupportedInB2T: emptyStringArray,
+        AspectsSupportedInT2B: ['all', 'depth-only'],
+      },
+      stencil8: {
+        AspectsSupportedInB2T: ['all', 'stencil-only'],
+        AspectsSupportedInT2B: ['all', 'stencil-only'],
+      },
+    };
+
+    {
+      const success = kDepthStencilFormatCapabilityInBufferTextureCopy[
+        format
+      ].AspectsSupportedInB2T.includes(aspect);
```

Comment:
Consider encapsulating the data table above, plus this check, into a function e.g.
`depthStencilCopySupported(type: 'CopyB2T' | 'CopyT2B', format: DepthStencilFormat, aspect: GPUTextureAspect)`
or possibly two functions (with either one shared table or two separate tables):
`depthStencilCopyB2TSupported(format: DepthStencilFormat, aspect: GPUTextureAspect)`
`depthStencilCopyT2BSupported(format: DepthStencilFormat, aspect: GPUTextureAspect)`

This can live in `capability_info` (and please document it in the `helper_index`). (Someday we could consider if it should be moved, or described as a table instead of a function, or generalized to non-depth-stencil formats too, but I think this is the most straightforward thing right now.)